### PR TITLE
Common build deploy

### PR DESCRIPTION
**Pull request by 2uasimojo**
/hold

DO NOT MERGE

The first commit needs to go into (and be updated from) boilerplate.

The second commit will be rebased onto that boilerplate update.

This is proof of concept.

### CODE REVIEW

File: boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh [hunk 1]
```diff
@@ -32,13 +35,13 @@ if [ ! -d "${SAAS_OPERATOR_DIR}/.git" ] ; then
 fi
 
 cat <<EOF > $DOCKERFILE_REGISTRY
-FROM quay.io/openshift/origin-operator-registry:latest
+FROM quay.io/openshift/origin-operator-registry:4.8.0
```
Comment (jharrington22) [inline boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh:38]: :pray: - though this is the release that OLM reported pushing a regression. I tested `latest` it was ok but I also tested `4.7` which was fine and that matches our OCP version, go with that.

File: boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh [hunk 2]
```diff
@@ -2,16 +2,20 @@
 
 source `dirname $0`/common.sh
 
-usage() { echo "Usage: $0 -o operator_name -c saas-repository-channel" 1>&2; exit 1; }
+usage() { echo "Usage: $0 -o operator-name -c saas-repository-channel -r registry-image" 1>&2; exit 1; }
```
Comment (jharrington22) [inline boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh:5]: `registry-image` for local testing? I thought about adding the same thing.

**Review by jharrington22 — COMMENTED** [review]
This is much neater than my example. :+1:

**Comment by jharrington22** [comment]
@2uasimojo what does the bundle dir look like compared to whats in https://gitlab.cee.redhat.com/service/saas-deadmanssnitch-operator-bundle? Same artifacts?

This operator has the expected `deploy/role.yaml` and `deploy/operator.yaml` files.

File: boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh (hunk 2 above)
Comment (2uasimojo) [reply boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh:5]: Yes, this allowed me to remove hardcoding that uri so I could set the environment variable from my shell.

**Comment by 2uasimojo** [comment]
> @2uasimojo what does the bundle dir look like compared to whats in https://gitlab.cee.redhat.com/service/saas-deadmanssnitch-operator-bundle? Same artifacts?
> 
> This operator has the expected `deploy/role.yaml` and `deploy/operator.yaml` files.

I didn’t get a chance to look at all that yet. Will do it in the morning.

File: boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh (hunk 1 above)
Comment (2uasimojo) [reply boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh:38]: Ack, will do

Comment (2uasimojo) [reply boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh:38]: Done.

File: boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh (hunk 2 above)
Comment (2uasimojo) [reply boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh:5]: I've put together a full guide for local testing [here](https://github.com/openshift/boilerplate/blob/eb067b23871d77cefb8eca1f7ea2c0f51cd1d719/boilerplate/openshift/golang-osd-operator/app-sre.md). Enjoy.

**Comment by 2uasimojo** [comment]
The real thing is at #108

**Pull request closed without merging**